tests/tests/lock-shrink-read-race.sh

```diff
@@ -0,0 +1,38 @@
+#
+# We had a lock server refcounting bug that could let one thread get a
+# reference on a lock struct that was being freed by another thread.  We
+# were able to reproduce this by having all clients try and produce a
+# lot of read and null requests.
+#
+# This will manfiest as a hung lock and timed out test runs, probably
+# with hung task messages on the console.
```

Comment:
I would add the comment that it will trigger KASAN when it's enabled. Perhaps with a fingerprint of the KASAN output.

Reply:
Fair enough.  I added a note, though it's not guaranteed, and mentioned the function it'll be found in.  (also fixed some silly trailing whitespace in the golden output).